crates/router/src/core/payment_methods/network_tokenization.rs

```
@@ -0,0 +1,739 @@
+                    developer_message = %parsed_error.error_info.developer_message,
+                    "Network tokenization error: {}",
+                    parsed_error.error_message
+                );
+                Err(errors::NetworkTokenizationError::ResponseDeserializationFailed)
+                    .attach_printable(format!("Response Deserialization Failed: {err_res:?}"))
+            }
+            Ok(res) => Ok(res),
+        })
+        .inspect_err(|err| {
+            logger::error!("Error while deserializing response: {:?}", err);
+        })?;
+
+    let token_response: TokenResponse = res
+        .response
+        .parse_struct("Get Network Token Response")
+        .change_context(errors::NetworkTokenizationError::ResponseDeserializationFailed)?;
+    logger::info!("Fetch Network Token Response: {:?}", token_response);
+
+    Ok(token_response)
+}
+
+pub async fn get_token_from_tokenization_service(
+    state: &routes::SessionState,
+    key_store: &domain::MerchantKeyStore,
+    network_token_requestor_ref_id: String,
+    pm_data: &storage::PaymentMethod,
+) -> errors::RouterResult<NetworkTokenData> {
+    let token_response =
+        if let Some(network_tokenization_service) = &state.conf.network_tokenization_service {
+            get_network_token(
+            state,
+            pm_data.customer_id.clone(),
+            network_token_requestor_ref_id,
+            network_tokenization_service.get_inner(),
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Fetch network token failed")
+        .inspect_err(
+            |e| logger::error!(error=?e, "Error while fetching token from tokenization service"),
+        )
+        } else {
+            Err(errors::NetworkTokenizationError::NetworkTokenizationServiceNotConfigured)
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .inspect_err(|_| {
```

Don't discard the error, log the error too